### PR TITLE
ci: add github workflow

### PR DESCRIPTION
**Pull request by x1ddos**
Alternative to Travis CI.
Runs the same scripts except SSH for nightly builds.

### CODE REVIEW

File: frontends/qt/Makefile [hunk 1]
```diff
@@ -52,7 +52,7 @@ osx:
 	mv build/BitBox.app build/osx/
 	cp resources/MacOS/Info.plist build/osx/BitBox.app/Contents/
 	cp resources/MacOS/icon.icns build/osx/BitBox.app/Contents/Resources/
-	macdeployqt build/osx/BitBox.app/
```
Comment (x1ddos) [inline frontends/qt/Makefile:55]: The spurious trailing slash was causing macdeployqt fail on github macos runner.
Not sure where it was coming from but it doesn't seem to be required according to https://doc.qt.io/qt-5/macos-deployment.html.

**Comment by benma** [comment]
Can you explain again what is going on with the vendoring?

Cause `make go-vendor` does not change anything when I run it locally.

Is Go 1.14 forcing a dep update?

If so, let's update Go properly in the Dockerfile?

**Comment by benma** [comment]
> If so, let's update Go properly in the Dockerfile?

Tried doing that here (might be independent, but was about time anyway):

https://github.com/digitalbitbox/bitbox-wallet-app/pull/850

**Comment by x1ddos** [comment]
I remember there was a change in Go 1.14 which was not backported to 1.13. The change made `go mod vendor` add all transitive deps where 1.13 would sometimes skip it. This results in 1.14 arriving at different deps versions where 1.13 just wouldn't care and skip. This is my understanding. It's buried somewhere in https://github.com/golang/go/issues/33848#issuecomment-536088704. Release notes https://golang.org/doc/go1.14#tools don't mention this inconsistency either.

> Cause make go-vendor does not change anything when I run it locally.

Yeah, I would expect exactly that on 1.13.

> Is Go 1.14 forcing a dep update?

Yes. It complains otherwise,

> If so, let's update Go properly in the Dockerfile?

Right, it's independent. I would expect updated deps to work with both 1.13 and 1.14.

**Comment by x1ddos** [comment]
I'll wait for #850 and then rebase. Feels like deps sync belongs more to that than this PR.

**Comment by x1ddos** [comment]
Ok. Removed the sync deps commit but forgot to sign.

**Pull request merged**